src/components/SponsorPage/SponsorPage.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useContext, useEffect, useState, useRef } from "react";
+import { useLocation } from "react-router";
+import s from "./SponsorPage.module.scss";
+import { Button, ModalWindow } from "../../shared/components";
+import { useNavigate } from "react-router-dom";
+import { TalentsService } from "../../services/api-services";
+import { UserContext } from "../../context/UserContext";
+import { useCookies } from "react-cookie";
+import { SponsorData } from "./components/SponsorData";
+export function SponsorPage() {
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    // const talentDataRef = useRef(null);
+    // const aboutRef = useRef(null);
+
+    const { user, token } = useContext(UserContext);
+    const [profile, setProfile] = useState({});
+    const [isLoading, setIsLoading] = useState(false);
+    // const [cookies, setCookie, removeCookie] = useCookies(["token", "user"]);
+
+    const [editting, setEditting] = useState(false);
+    // const { setUserInfo } = useContext(UserContext);
+    // const [modalIsOpen, setModalIsOpen] = useState(false);
+    // const [cancelModalIsOpen, setCancelModalIsOpen] = useState(false);
+    // const [saveError, setSaveError] = useState("");
+
+    useEffect(() => {
+        if (editting) {
+            navigate(`${location.pathname}${location.search}#edit`, {
+                replace: true,
+            });
+        } else if (!editting) {
+            navigate(`${location.pathname}${location.search}`, {
+                replace: true,
+            });
+        }
```

Comment:
Too long , rewrite it like:
const path = `${location.pathname}${location.search}`;
const hash = editting ? '#edit' : '';
navigate(`${path}${hash}`, { replace: true });